plugins/modules/identity/bitwarden.py

```diff
@@ -0,0 +1,371 @@
+        type: str
+        description:
+            - Name of item to target.
+            - Mutually exclusive from I(item_id).
+    item_id:
+        type: str
+        description:
+            - UID of item to target.
+            - Mutually exclusive from I(item_name).
+    login:
+        type: dict
+        description:
+            - Mapping of keys to values for C(bw) login details such as C(username) and C(password).
+        default: {}
+    notes:
+        type: str
+        description:
+            - A note about the item that is stored as metadata about the item.
+    target:
+        type: str
+        description:
+            - The type of item being added/modified/removed/queried.
+        choices:
+            - item
+            - folder
+        default: 'item'
+    state:
+        type: str
+        description:
+            - Action to perform on target.
+            - The C(absent) option will remove the target if it exists. This includes non-empty folders.
+            - The C(created) option will create the target, if needed, with the fields in I(login) and I(notes).
+            - C(created) will not update an existing entry.
+            - The C(present) option will create the target, if needed, and update fields in I(login) and I(notes).
+        choices:
+            - absent
+            - created
+            - present
+        default: 'present'
+'''
+
+EXAMPLES = '''
+- name: "Ensure a folder exists"
+  community.general.bitwarden:
+    folder_name: "my_folder"
+    target: "folder"
+    state: "present"
+
+- name: "Ensure an entry exists in the database"
+  community.general.bitwarden:
+    folder_name: "my_folder"
+    item_name: "some item"
+    login:
+      username: "a_user"
+      password: "some password"
+    notes: >-
+      Some information about a user is stored here. foo
+    target: "item"
+    state: "present"
+  delegate_to: "localhost"
+
+- name: "Ensure an entry does not exist in the database"
+  community.general.bitwarden:
+    folder_name: "my_folder"
+    item_name: "some item"
+    target: "item"
+    state: "absent"
+  delegate_to: "localhost"
+
+- name: "Create an entry in the database if it does not already exist"
+  community.general.bitwarden:
+    folder_name: "my_folder"
+    item_name: "some item"
+    login:
+      username: "a_user"
+      password: "some password"
+    notes: >-
+      Some information about a user is stored here. foo
+    target: "item"
+    state: "created"
+  delegate_to: "localhost"
+'''
+
+RETURN = '''
+---
+# A dictionary with the new/queried state.
+'''
+
+
+from collections import defaultdict
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.community.general.plugins.module_utils.identity.bitwarden import (
+    BitwardenException, Client, Organization
+)
+
+
+class Bitwarden(object):
+    '''Accessor for mutating Bitwarden vaults.'''
+
+    organization = property(lambda self: self._organization)
```

Comment:
There are two primary reasons I opted to use a number of read-only properties, both as defences against accidents made during future updates/calls to the modules in Python since the model itself isn't actually very mutable (since the objects are only _views_ into data in the Bitwarden vault and not an in-memory Python model which I agree _would_ be much more suited to direct attribute access).

Firstly, given there is persistence, I wanted to make it a bit less likely someone would think that changing the model's value would update the persistent store.

More importantly, I wanted to enforce data consistency. For example, in the constructors for different `_Element`s, specifying both names and IDs is forbidden; allowing direct mutation outside of the class would either require additional setters to do validation or we would have to accept the data could be in an inconsistent state. Another example of inconsistent data that could arise is if an item's organization were changed without changing the folder (this is _sometimes_ possibly legal in Bitwarden since some folders can exist in multiple organizations). We could then end up with an object that contains real data but does not (and cannot) exist anywhere in a real vault. Making attributes read-only was an easy way out of having to worry about object consistency and introducing even more setters.